tests/test_workers/test_pkg_managers/test_gomod.py

```diff
@@ -266,8 +266,7 @@ def test_resolve_gomod(
     if dep_replacement:
         run_side_effects.append(mock.Mock(returncode=0, stdout=None))  # go mod edit -replace
     run_side_effects.append(mock.Mock(returncode=0, stdout=None))  # go mod download
-    if dep_replacement:
-        run_side_effects.append(mock.Mock(returncode=0, stdout=None))  # go mod tidy
```

Comment:
Maybe it's failing because you don't need to run `go mod tidy` always? Only when the list `dep_replacement` is set

Reply:
Oh, I see it's a part of the story, I am sorry, just blind guess

Reply:
For reference, these are the test cases:
https://github.com/containerbuildsystem/cachito/runs/4815799205?check_suite_focus=true#step:6:119
https://github.com/containerbuildsystem/cachito/runs/4815799205?check_suite_focus=true#step:6:213

Reply:
As far as I understand, using `go mod tidy` will make changes to the go.mod file, and that behavior might not be wanted in all cases.  

https://go.dev/ref/mod#go-mod-tidy

So, I think we should only run `go mod tidy` in cases the new flag is present (and the current cases in which it is already run, `go-mod-vendor` if I'm not mistaken). This way, we keep backwards compatibility.